provider/cluster_autoscaler_resource.go

```diff
@@ -0,0 +1,703 @@
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+}
+
+func (r *ClusterAutoscalerResource) Update(ctx context.Context, request tfsdk.UpdateResourceRequest,
+	response *tfsdk.UpdateResourceResponse) {
+	var diags diag.Diagnostics
+
+	// Get the plan:
+	plan := &ClusterAutoscalerState{}
+	diags = request.Plan.Get(ctx, plan)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	_, err := r.collection.Cluster(plan.Cluster.Value).Autoscaler().Get().SendContext(ctx)
+
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed getting cluster autoscaler",
+			fmt.Sprintf(
+				"Failed getting autoscaler for cluster '%s': %v",
+				plan.Cluster.Value, err,
+			),
+		)
+		return
+	}
+
+	autoscaler, err := clusterAutoscalerStateToObject(plan)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed updating cluster autoscaler",
+			fmt.Sprintf(
+				"Failed updating autoscaler for cluster '%s: %v ",
+				plan.Cluster.Value, err,
+			),
+		)
+		return
+	}
+
+	update, err := r.collection.Cluster(plan.Cluster.Value).
+		Autoscaler().Update().Body(autoscaler).SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed updating cluster autoscaler",
+			fmt.Sprintf(
+				"Failed updating autoscaler for cluster '%s': %v",
+				plan.Cluster.Value, err,
+			),
+		)
+		return
+	}
+
+	object := update.Body()
+	state := &ClusterAutoscalerState{}
+	populateAutoscalerState(object, plan.Cluster.Value, state)
+
+	diags = response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r *ClusterAutoscalerResource) Delete(ctx context.Context, request tfsdk.DeleteResourceRequest,
+	response *tfsdk.DeleteResourceResponse) {
+	state := &ClusterAutoscalerState{}
+	diags := request.State.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+
+	resource := r.collection.Cluster(state.Cluster.Value).Autoscaler()
+	_, err := resource.Delete().SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed deleting cluster autoscaler",
+			fmt.Sprintf(
+				"Failed deleting autoscaler for cluster '%s': %v",
+				state.Cluster.Value, err,
+			),
+		)
+		return
+	}
+
+	response.State.RemoveResource(ctx)
+}
+
+func (r *ClusterAutoscalerResource) ImportState(ctx context.Context, request tfsdk.ImportResourceStateRequest,
+	response *tfsdk.ImportResourceStateResponse) {
+	clusterId := request.ID
+
+	getResponse, err := r.collection.Cluster(clusterId).Autoscaler().Get().SendContext(ctx)
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Failed importing cluster autoscaler",
+			err.Error(),
+		)
+		return
+	}
+
+	state := &ClusterAutoscalerState{}
+	populateAutoscalerState(getResponse.Body(), clusterId, state)
+	diags := response.State.Set(ctx, state)
+	response.Diagnostics.Append(diags...)
+}
```

Comment:
Can be more than one autoscaler for a cluster? 
I guess it's only one per cluster (otherwise you should set an identifier), and if this is the case you can use the "simple" import state 
```suggestion
	func (r *ClusterRosaClassicResource) ImportState(ctx context.Context, request resource.ImportStateRequest,
	response *resource.ImportStateResponse) {
	tflog.Debug(ctx, "begin importstate()")

	resource.ImportStatePassthroughID(ctx, path.Root("cluster"), request, response)
}

```